dirty_cat/_super_vectorizer.py

```diff
@@ -4,7 +4,6 @@
 manually categorize them beforehand, or construct complex Pipelines.
 """
 
-
 from typing import Dict, List, Literal, Optional, Tuple, Union
```

Comment:
Ok, I see you removed here and everywhere else the `typing.Optional` (and maybe some other ones as well) and put directly the type name. If that's the case you should also remove the imports associated. To be consistent, if this is the way to go, we should remove `typing.Optional` in other indicators as well?
```suggestion
from typing import Dict, List, Literal, Tuple, Union
```

Reply:
Yes I removed these occurrences in the docstring, but not in the code.
For example `typing.Optional` is still used [here](https://github.com/dirty-cat/dirty_cat/blob/d634f0e3845e31414e25260d764a67d456d2f231/dirty_cat/_super_vectorizer.py#L81). Also, if we had unused imports, the `Run pre-commit checks` test would fail.
The point of this PR is to make the [generated doc](https://output.circle-artifacts.com/output/job/7d81266c-bf80-49e6-90fc-68f6855743d1/artifacts/0/doc/generated/dirty_cat.SuperVectorizer.html#dirty_cat.SuperVectorizer) cleaner by formatting the docstring to the numpydoc guidelines.

Reply:
Looking back at the numpy style guide, I noticed this is not the expected way of doing it, so I modified it :)